datasetinsights/io/source/base.py

```diff
@@ -0,0 +1,42 @@
+import re
```

Comment:
Would it be better to have `datasetinsights/io/downloader` as everything in this module will be related to different downloader?

Reply:
I think it might make sense, but there is a module called downloader.py in datasetinsights/io/ already so we might need to move this or rename it

Reply:
I also agree with @adason with having a `downloader` subdirectory in `io`. Also, @AdamPalmarUnity the module is `download.py`.

Reply:
I'm fine with naming the subdirectory downloader or dataset_downloader.